project/eScavenge/Team.py

```diff
@@ -1,95 +1,56 @@
 import unittest
-from abc import ABC, abstractmethod
-from datetime import timedelta
+from datetime import timedelta, datetime
+from django.db import models
 
-from .GameMaker import UserABC
 
+class Team(models.Model):
+    username = models.CharField(max_length=20, unique=True)
+    __password = models.CharField(max_length=20, db_column="password")
+    __points = models.IntegerField(default=0)
+    current_landmark = models.IntegerField(default=0)
+    penalty_count = models.IntegerField(default=0)
+    clue_time = models.DateTimeField(default=datetime.now())
 
-class TeamI(ABC):
-    @abstractmethod
-    def login(self, username, password):
-        pass
+    def __eq__(self, other):
+        return self.username == other.username
 
-    @abstractmethod
-    def add_penalty(self, penalty):
-        pass
+    def login(self, username, password):
+        if username != self.username or password != self.__password:
+            return None
+        return self
 
     @property
-    @abstractmethod
     def points(self):
-        pass
+        return self.__points
 
     @points.setter
-    @abstractmethod
     def points(self, points):
-        pass
+        self.__points = max(0, points)
+
+    def add_penalty(self, penalty=1):
```

Comment:
Are we assuming now that the penalty function will just be called and the value will always be one?

Reply:
It defaults to one, you can override it on call to whatever you want

Reply:
So add_penalty() will add 1, add_penalty(5) will add 5

Reply:
Gotcha okay, that makes sense.  I assumed Python always defaulted to that value.